src/imgix-core-js.js

```diff
@@ -59,9 +81,18 @@
         queryParams = this._signParams(path, queryParams);
       }
 
-      return this.settings.urlPrefix + this.settings.host + path + queryParams;
+      return this.settings.urlPrefix + this._getDomain(path) + path + queryParams;
     };
 
+    ImgixClient.prototype._getDomain = function(path) {
+      if(this.settings.shard_strategy === SHARD_STRATEGY_CYCLE) {
+        var domain = this.settings.domains[this._shard_next_index];
+        this._shard_next_index = (this._shard_next_index + 1) % this.settings.domains.length;
+        return domain;
+      }
+      return this.settings.domains[crc.crc32(path) % this.settings.domains.length];
```

Comment:
Can we wrap this line in an `else if (this.settings.shard_strategy === SHARD_STRATEGY_CRC) { ... }`? I'd rather be verbose than obscure.